handlers/backend_handler.go

```diff
@@ -56,22 +57,47 @@ type backendTransport struct {
 // Construct a backendTransport that wraps an http.Transport and implements http.RoundTripper.
 // This allows us to intercept the response from the backend and modify it before it's copied
 // back to the client.
-func newBackendTransport(connectTimeout, headerTimeout time.Duration, logger logger.Logger) (transport *backendTransport) {
-	transport = &backendTransport{&http.Transport{}, logger}
+func newBackendTransport(connectTimeout, headerTimeout time.Duration, logger logger.Logger) *backendTransport {
```

Comment:
Is there a reason not to consume the context? ... think if you create your own net.Dailer you can use the DialContext method to honour the context... but not sure if it is intentional or not

```
dialer := &net.Dialer{Timeout: connectTimeout}
transport.DialContext = func(ctx context.Context, network, address string) (net.Conn, error) {
	return dialer.DialContext(ctx, network, address)
}
```

Reply:
You have a very good point

I did it this way because ignoring the context is equivalent to the behaviour before, just using a non-deprecated function.

I guess what would be better is your solution but also configuring the `Dialer` to have good defaults other than the Timeout:

```
    DialContext: (&net.Dialer{
        Timeout:   connectTimeout,   // Configured by caller
        KeepAlive: 30 * time.Second, // same as DefaultTransport
        DualStack: true,             // same as DefaultTransport
    }).DialContext,
```